Avoid pattern mismatch, add option to specify escape character

GKJCJG opened the pull request:
There are cases in which :...: may occur in, for example, a URL where inserting an emoji would break a link. This adds an option to the plugin to replace only if the emoji pattern is surrounded by whitespace. This should be false by default, as it incurs a performance penalty.

Notes on the code: using the boundary assertion (\b) would be more performant but doesn't work for emoji patterns because : is a non-word character. This implementation uses lookbehinds, which are relatively new to Node. An alternative is to replace the lookbehind with `([\s\n])` and add `$1` to the `replacement` if `requireWhiteSpace` is `true`.

Commit:
Add word break option.

matchilling committed:
Add CircleCI config

matchilling commented:
Hi Colin 👋, thanks for taking the time and raising the issue. While I think your proposed solution solves the problem in a very pragmatic way, I would like your feedback on a different idea.

From the usability perspective of an editor, I find it hard to spot these additional whitespaces in the content. Therefore I would like instead to be able to use a unique escape character which is easier to distinguish from the rest of the content.

For example, we could add the following option to the plugin:

```js
{
  // In order to avoid pattern mismatch you can specify
  // an escape character which will be prepended to the
  // actual pattern (e.g. `#:poop:`).
  escapeCharacter : '#' // (default: '')
}
```

What do you think @GKJCJG?

matchilling committed:
Add "escapeCharacter" option to avoid pattern mismatch

{
  // In order to avoid pattern mismatch you can specify
  // an escape character which will be prepended to the
  // actual pattern (e.g. `#:poop:`).
  escapeCharacter : '#' // (default: '')
}

GKJCJG commented:
I think that is a good idea that incurs less of a performance penalty. Depending on how worried you are about mismatch, you can then specify as complex an escape as necessary :).
